// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: envoy/service/discovery/v2/mist_sign.proto

package envoy_service_discovery_v2

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"strings"
	"time"
	"unicode/utf8"

	"github.com/golang/protobuf/ptypes"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = ptypes.DynamicAny{}
)

// define the regex for a UUID once up-front
var _mist_sign_uuidPattern = regexp.MustCompile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$")

// Validate checks the field values on MistSignRequest with the rules defined
// in the proto definition for this message. If any rules are violated, an
// error is returned.
func (m *MistSignRequest) Validate() error {
	if m == nil {
		return nil
	}

	// no validation rules for SignString

	for idx, item := range m.GetExt() {
		_, _ = idx, item

		if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return MistSignRequestValidationError{
					field:  fmt.Sprintf("Ext[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	return nil
}

// MistSignRequestValidationError is the validation error returned by
// MistSignRequest.Validate if the designated constraints aren't met.
type MistSignRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e MistSignRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e MistSignRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e MistSignRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e MistSignRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e MistSignRequestValidationError) ErrorName() string { return "MistSignRequestValidationError" }

// Error satisfies the builtin error interface
func (e MistSignRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sMistSignRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = MistSignRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = MistSignRequestValidationError{}

// Validate checks the field values on MistSignResponse with the rules defined
// in the proto definition for this message. If any rules are violated, an
// error is returned.
func (m *MistSignResponse) Validate() error {
	if m == nil {
		return nil
	}

	// no validation rules for Signature

	for idx, item := range m.GetExt() {
		_, _ = idx, item

		if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return MistSignResponseValidationError{
					field:  fmt.Sprintf("Ext[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	return nil
}

// MistSignResponseValidationError is the validation error returned by
// MistSignResponse.Validate if the designated constraints aren't met.
type MistSignResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e MistSignResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e MistSignResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e MistSignResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e MistSignResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e MistSignResponseValidationError) ErrorName() string { return "MistSignResponseValidationError" }

// Error satisfies the builtin error interface
func (e MistSignResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sMistSignResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = MistSignResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = MistSignResponseValidationError{}
